includes/class-llms-awards-query.php

```diff
@@ -0,0 +1,525 @@
+	}
+
+	/**
+	 * Determine if any results were found for the query.
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean
+	 */
+	public function has_results() {
+		return count( $this->query->posts ) > 0;
+	}
+
+	/**
+	 * Prepares the `meta_query` ultimately passed to the WP_Query.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $args Cleaned LLMS_Awards_Query arguments.
+	 * @return array An array of meta query arrays.
+	 */
+	private function prepare_meta_query( $args ) {
+
+		// If a query for manual awards we skip all other relationships and return early.
+		if ( $args['manual'] ) {
+
+			return array(
+				'relation' => 'OR',
+				// Meta doesn't exist.
+				array(
+					'key'     => '_llms_engagement',
+					'compare' => 'NOT EXISTS',
+				),
+				// Or it's "empty".
+				array(
+					'key'     => '_llms_engagement',
+					'value'   => array( '', '0', 0 ),
+					'compare' => 'IN',
+				),
+			);
+
+		}
+
+		return $this->prepare_meta_query_for_relationships( $args );
+
+	}
+
+	/**
+	 * Retrieve meta query parts for related posts, engagements, and templates.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $args Cleaned LLMS_Awards_Query arguments.
+	 * @return array An array of meta query arrays.
+	 */
+	private function prepare_meta_query_for_relationships( $args ) {
+
+		$meta_query = array();
+
+		$relations = array(
+			'related_posts' => '_llms_related',
+			'engagements'   => '_llms_engagement',
+			'templates'     => $this->get_template_meta_key(),
+		);
+		foreach ( $relations as $arg => $meta_key ) {
+
+			// Include.
+			if ( ! empty( $args[ $arg ] ) ) {
+				$meta_query[] = array(
+					'key'     => $meta_key,
+					'value'   => $this->to_int_array( $args[ $arg ] ),
+					'compare' => 'IN',
+				);
+			}
+
+			// Exclude.
+			$exclude_arg = $arg . '__exclude';
+			if ( ! empty( $args[ $exclude_arg ] ) ) {
+
+				$meta_query[] = array(
+					'relation' => 'OR',
+					array(
+						'key'     => $meta_key,
+						'value'   => $this->to_int_array( $args[ $exclude_arg ] ),
+						'compare' => 'NOT IN',
+					),
+					// Ensure posts that don't have the metadata set will be returned.
+					array(
+						'key'     => $meta_key,
+						'compare' => 'NOT EXISTS',
+					),
+				);
+			}
+		}
+
+		return $meta_query;
+
+	}
+
+	/**
+	 * Prepare the WP_Query arguments for the awards query.
+	 *
+	 * @since [version]
+	 *
+	 * @param array $args Input arguments from `__construct()`.
+	 * @return array Array of arguments suitable to pass to a WP_Query.
+	 */
+	private function prepare_query( $args ) {
+
+		$args = $this->clean_args( wp_parse_args( $args, $this->get_defaults() ) );
+
+		$query_args = array(
+			'author__in'     => $args['users'],
+			'author__not_in' => $args['users__exclude'],
+			'post_type'      => $this->get_post_type(),
+			'paged'          => $args['page'],
+			'per_page'       => $args['per_page'],
+			'no_found_rows'  => $args['no_found_rows'],
+			'orderby'        => $args['sort'],
+		);
```

Comment:
We need a `post_status` arg too:
https://developer.wordpress.org/reference/classes/wp_query/#status-parameters

Basically when not in admin the `post_status` defaults to only `publish` but we want to fetch also scheduled awarded certificates: we're not in admin when doing background processors.

Reply:
yea I saw, the `status` arg.